File: backend/src/interface/models/HouseDetail.ts
import { ObjectId } from "mongoose";

export default interface HouseDetail {
	house_id: ObjectId;
	user_id: ObjectId;
	rooms: [
		{
			type: string;
			amount: number;
			pictures: [
				{
					url: string;
				}
			];
		}
	];
	description: string;
	facilities: [
		{
			name: string;
			checked: boolean;
		}
	];
	electric_fee: number;
	water_fee: number;
	likes: number;
	total_size: number;
}

File: besthouse/lib/models/house_detail.dart
import 'package:besthouse/models/room.dart';

class HouseDetail {
  final String houseId;
  final String userId;
  final List<Room> rooms;
  final List<String> facilities;
  final String description;
  final double electricFee;
  final int likes;
  final double totalSize;

  HouseDetail({
    required this.houseId,
    required this.userId,
    required this.rooms,
    required this.facilities,
    this.description = "",
    this.electricFee = 0.0,
    this.likes = 0,
    this.totalSize = 0.0,
  });
}
